src/Main.cpp

```diff
@@ -111,24 +113,46 @@ struct EnemyMovementSystem
 
 	void on_tick() override
 	{
-		
-		
-		for (auto &&[player_entity, player_pos] : QueryPlayers::access_storage().each())
+		//enemy flocking system
+
+		auto player  = QueryPlayers::access_storage().front();
+		for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
 		{
-			for (auto&& [entity, enemy, pos] : QueryEnemies::access_storage().each())
+			auto min_distance = 1000000;
+			auto nearest_enemy = Enemy();
+			auto nearest_enemy_pos = Position();
+			for (auto&& [entity2, enemy2, pos2] : QueryEnemies::access_storage().each())
 			{
-				if (pos.xy.y >= SCREEN_HEIGHT - BALL_RADIUS || pos.xy.y <= BALL_RADIUS)
+				if(entity != entity2)
 				{
-					enemy.speed.y *= -1;
+					auto dist = distance(pos.xy, pos2.xy);
+					if (dist < min_distance)
+					{
+						min_distance = dist;
+						nearest_enemy = enemy2;
+						nearest_enemy_pos = pos2;
+					}
 				}
+			}
+			auto player_position = MutAccessComponentById<Position>::get(player);
+			auto to_player =  normalize(player_position.xy - pos.xy);
+			
+			auto from_nearest_enemy  = normalize(pos.xy-nearest_enemy_pos.xy);
```

Comment:
Are we doing normal distance calculations here? Can't see where normalize is defined but if we're not doing anything out of the ordinary, let's just have a function that says `distance` and keep it simple?

Reply:
At first, we defined our own functions distance and normalize. Then we got multiple definitions error because they were already defined  somewhere in subproject/glm/...
Without normalizing vectors, enemies get out of range very quickly and it is much harder to adjust distances between them by changing coefficients (because vector intensity is playing big role).